src/AdminConsole/Services/PasswordlessManagement/PasswordlessManagementClient.cs

```diff
@@ -2,126 +2,160 @@
 
 namespace Passwordless.AdminConsole.Services.PasswordlessManagement;
 
-public class PasswordlessManagementClient : IPasswordlessManagementClient
+public class PasswordlessManagementClient(HttpClient http) : IPasswordlessManagementClient
 {
-    private readonly HttpClient _client;
-
-    public PasswordlessManagementClient(HttpClient client)
+    public async Task<CreateAppResultDto> CreateApplicationAsync(string appId, CreateAppDto options)
     {
-        _client = client;
-    }
+        using var response = await http.PostAsJsonAsync(
+            $"/admin/apps/{Uri.EscapeDataString(appId)}/create",
+            options
+        );
 
-    public async Task<CreateAppResultDto> CreateApplication(string appId, CreateAppDto options)
-    {
-        var res = await _client.PostAsJsonAsync($"/admin/apps/{appId}/create", options);
-        res.EnsureSuccessStatusCode();
-        return await res.Content.ReadFromJsonAsync<CreateAppResultDto>();
+        response.EnsureSuccessStatusCode();
+
+        return await response.Content.ReadFromJsonAsync<CreateAppResultDto>();
     }
 
-    public async Task<MarkDeleteApplicationResponse> MarkDeleteApplication(MarkDeleteApplicationRequest request)
+    public async Task<MarkDeleteApplicationResponse> MarkDeleteApplicationAsync(MarkDeleteApplicationRequest request)
     {
-        var response = await _client.PostAsJsonAsync($"admin/apps/{request.AppId}/mark-delete", new { request.DeletedBy });
+        using var response = await http.PostAsJsonAsync(
+            $"admin/apps/{Uri.EscapeDataString(request.AppId)}/mark-delete",
+            new { request.DeletedBy }
+        );
+
         response.EnsureSuccessStatusCode();
+
         return await response.Content.ReadFromJsonAsync<MarkDeleteApplicationResponse>();
     }
 
     public async Task<ICollection<string>> ListApplicationsPendingDeletionAsync()
     {
-        var response = await _client.GetAsync("apps/list-pending-deletion");
+        using var response = await http.GetAsync("apps/list-pending-deletion");
         response.EnsureSuccessStatusCode();
+
         return await response.Content.ReadFromJsonAsync<ICollection<string>>();
     }
 
-    public async Task<bool> DeleteApplicationAsync(string application)
+    public async Task<bool> DeleteApplicationAsync(string appId)
     {
-        var res = await _client.DeleteAsync($"admin/apps/{application}");
-        return res.IsSuccessStatusCode;
+        using var response = await http.DeleteAsync($"admin/apps/{Uri.EscapeDataString(appId)}");
+        // response.EnsureSuccessStatusCode(); ???
```

Comment:
Why are we not checking the status code here? Is it intentional or an oversight?